tensorflow_addons/layers/multihead_attention.py

```diff
@@ -0,0 +1,128 @@
+import tensorflow as tf
+
+
+class MultiHeadAttention(tf.keras.layers.Layer):
+    r"""
+    """
+
+    def __init__(
+        self,
+        head_size,
+        num_heads,
+        output_size=None,
+        dropout_rate=None,
+        use_projection_bias=True,
+        **kwargs
+    ):
+        self.head_size = head_size
+        self.num_heads = num_heads
+        self.output_size = output_size
+        self.dropout_rate = dropout_rate
+        self.use_projection_bias = use_projection_bias
+
+        super().__init__(**kwargs)
+
+    def build(self, input_shape):
+
+        num_query_features = input_shape[0][-1]
+        num_key_features = input_shape[1][-1]
+        num_value_features = input_shape[2][-1]
+        output_size = self.output_size if self.output_size else num_value_features
+
+        self.query_kernel = self.add_weight(
+            name="query_kernel",
+            shape=[self.num_heads, num_query_features, self.head_size],
+            # initializer=self.kernel_initializer,
+            # regularizer=self.kernel_regularizer,
+            # constraint=self.kernel_constraint,
+        )
+        self.key_kernel = self.add_weight(
+            name="key_kernel",
+            shape=[self.num_heads, num_key_features, self.head_size],
+            # initializer=self.kernel_initializer,
+            # regularizer=self.kernel_regularizer,
+            # constraint=self.kernel_constraint,
+        )
+        self.value_kernel = self.add_weight(
+            name="value_kernel",
+            shape=[self.num_heads, num_value_features, self.head_size],
+            # initializer=self.kernel_initializer,
+            # regularizer=self.kernel_regularizer,
+            # constraint=self.kernel_constraint,
+        )
+        self.projection_kernel = self.add_weight(
+            name="projection_kernel",
+            shape=[self.num_heads, self.size_per_head, output_size],
+            # initializer=self.kernel_initializer,
+            # regularizer=self.kernel_regularizer,
+            # constraint=self.kernel_constraint,
+        )
+
+        if self.use_projection_bias:
+            self.projection_bias = self.add_weight(
+                name="projection_bias",
+                shape=[output_size],
+                # initializer=self.bias_initializer,
+                # regularizer=self.bias_regularizer,
+                # constraint=self.bias_constraint,
+            )
+        else:
+            self.projection_bias = None
+
+        self.dropout = tf.keras.layers.Dropout(self.dropout_rate)
```

Comment:
The dropout layer should be used in the call method.

Reply:
Correct.